test/src/406-ducc-webhook-notifications/main

```diff
@@ -0,0 +1,139 @@
+cvmfs_test_name="Webhook notifications"
+cvmfs_test_autofs_on_startup=false
+cvmfs_test_suites="ducc"
+
+check_id_action() {
+    expected_id=$1
+    expected_action=$2
+
+    notification=$( tail -n 1 $notifications_file )
+    id=(${notification//|/ })
+
+    if [ ${id[0]} != ${expected_id} ]
+    then
+        echo 'The id of the action is not correct'
+        kill $pid
+        return 1
+    fi
+
+    if [ ${id[1]} != ${expected_action} ]
+    then
+        echo 'The action not correct'
+        kill $pid
+        return 2
+    fi
+
+    sleep 1
+}
+
+send_push() {
+    curl -X POST http://webhook-c8vm:8080/webhooks/test-cvmfs-ingestion -H "Content-Type: application/json" -d "{\"type\": \"PUSH_ARTIFACT\", \"event_data\": {\"resources\": [{\"resource_url\": \"registry.cern.ch/unpacked-dev/image:tag\"}]}}" || return 3
+}
+
+send_delete() {
+    curl -X POST http://webhook-c8vm:8080/webhooks/test-cvmfs-ingestion -H "Content-Type: application/json" -d "{\"type\": \"DELETE_ARTIFACT\", \"event_data\": {\"resources\": [{\"resource_url\": \"registry.cern.ch/unpacked-dev/image:tag\"}]}}" || return 4
+}
+
+send_replication() {
+    curl -X POST http://webhook-c8vm:8080/webhooks/test-cvmfs-ingestion -H "Content-Type: application/json" -d "{\"type\": \"REPLICATION\", \"event_data\": {\"replication\": { \"dest_resource\": {\"endpoint\": \"https://registry.cern.ch\", \"namespace\": \"unpacked-dev\"}, \"successful_artifact\": [{\"type\": \"image\", \"status\": \"Success\", \"name_tag\": \"image:tag\"}]}}}" || return 5
+}
+
+cvmfs_run_test() {
+
+    notifications_file="notifications.txt"
+    sudo python3 /root/cvmfs/test/src/406-ducc-webhook-notifications/registry-webhook.py -f $notifications_file -h 188.185.120.234 -p 8080 &
+
+    sleep 1
+    pid=$(echo $!)
+
+    yes | touch $notifications_file
+
+    send_push
+    check_id_action '0' 'push'
+
+    send_delete
+    check_id_action '1' 'delete'
+
+    send_replication
+    check_id_action '2' 'replication'
+
+    yes | touch $notifications_file
+
+    send_delete
+    check_id_action '0' 'delete'
+
+    send_replication
+    check_id_action '1' 'replication'
+
+    send_replication
+    check_id_action '2' 'replication'
+
+    send_push
+    check_id_action '3' 'push'
+
+    yes | touch $notifications_file
+
+    send_replication
+    check_id_action '0' 'replication'
+
+    send_push
+    check_id_action '1' 'push'
+
+    send_push
+    check_id_action '2' 'push'
+
+    send_push
+    check_id_action '3' 'push'
+
+    send_delete
+    check_id_action '4' 'delete'
+
+    send_delete
+    check_id_action '5' 'delete'
+
+    send_delete
+    check_id_action '6' 'delete'
+
+    echo "999|push|https://registry.cern.ch/unpacked-dev/image:tag" >> $notifications_file
+
+    send_delete
+    check_id_action '1000' 'delete'
+
+    original_content=$(<$notifications_file)
```

Comment:
Might be easier to just copy the file to a temporary one and than use `diff` to compare with the rotated.